spec/unit/provider/gluster_peer_spec.rb

```diff
@@ -1,15 +1,14 @@
 require 'spec_helper'
-require 'unit/provider/helpers'
-require 'unit/provider/fake_gluster'
+require 'unit/helpers'
 
-describe Puppet::Type.type(:gluster_peer).provider(:gluster_peer) do
+peer_type = Puppet::Type.type(:gluster_peer)
+
+describe peer_type.provider(:gluster_peer), :unit => true do
   on_supported_os.each do |os, facts|
     context "on #{os}" do
       before :each do
-        Facter.clear
-        facts.each do |k, v|
-          Facter.stubs(:fact).with(k).returns Facter.add(k) { setcode { v } }
-        end
+        allow(Facter).to receive(:collection, &make_fake_collection(facts))
+        Facter.reset
```

Comment:
Is there any way we can wrap this up as a helper method or something? I know it's only 2 lines but fairly complicated and repeated a few times.